From 1d76889c2cb290d5aeb0e67e8ecd29a9f6ae3b67 Mon Sep 17 00:00:00 2001
From: yifengyou <842056007@qq.com>
Date: Thu, 5 Mar 2026 04:41:17 +0000
Subject: [PATCH] adapt owl rk3588

Signed-off-by: yifengyou <842056007@qq.com>

diff --git a/arch/arm64/boot/dts/rockchip/Makefile b/arch/arm64/boot/dts/rockchip/Makefile
index 6826285852ad..eee14b1e55c7 100644
--- a/arch/arm64/boot/dts/rockchip/Makefile
+++ b/arch/arm64/boot/dts/rockchip/Makefile
@@ -38,6 +38,7 @@ dtb-$(CONFIG_CPU_RK3588) += rk3588-lubancat-5ioi.dtb
 dtb-$(CONFIG_CPU_RK3588) += rk3588s-lubancat-4.dtb
 dtb-$(CONFIG_CPU_RK3588) += rk3588s-lubancat-4-v1.dtb
 dtb-$(CONFIG_CPU_RK3588) += rk3588s-lubancat-4io.dtb
+dtb-$(CONFIG_CPU_RK3588) += rk3588-owl-ai-box-plus-v10.dtb
 
 subdir-y	:= $(dts-dirs) overlay
 
diff --git a/arch/arm64/boot/dts/rockchip/rk3588-owl-ai-box-plus-v10.dts b/arch/arm64/boot/dts/rockchip/rk3588-owl-ai-box-plus-v10.dts
new file mode 100644
index 000000000000..99ab413502c3
--- /dev/null
+++ b/arch/arm64/boot/dts/rockchip/rk3588-owl-ai-box-plus-v10.dts
@@ -0,0 +1,45 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+* Copyright (c) 2021 Rockchip Electronics Co., Ltd.
+*
+*/
+
+/dts-v1/;
+
+#include "rk3588-evb1-lp4.dtsi"
+#include "rk3588-evb1-imx415.dtsi"
+#include "rk3588-linux.dtsi"
+
+/ {
+	model = "OWL RK3588 AIBOX Plus V10 Board";
+	compatible = "rockchip,rk3588-evb1-lp4-v10", "rockchip,rk3588";
+
+	vbus5v_pwr: vbus5v-pwr {
+		compatible = "regulator-fixed";
+		regulator-name = "vbus5v_pwr";
+		gpio = <&gpio1 RK_PC6 GPIO_ACTIVE_HIGH>;
+		enable-active-high;
+		regulator-always-on;
+		regulator-boot-on;
+		regulator-min-microvolt = <5000000>;
+		regulator-max-microvolt = <5000000>;
+	};
+};
+
+&vcc_1v8_s0 {
+	/delete-property/ regulator-state-mem;
+	regulator-state-mem {
+		regulator-on-in-suspend;
+		regulator-suspend-microvolt = <1800000>;
+	};
+};
+
+&vcc_3v3_s0 {
+	/delete-property/ regulator-state-mem;
+	regulator-state-mem {
+		regulator-on-in-suspend;
+		regulator-suspend-microvolt = <3300000>;
+	};
+};
+
+
-- 
2.43.0

