From 0f00696babd1ea702aee71c7df5752c67ba9f458 Mon Sep 17 00:00:00 2001
From: yifengyou <842056007@qq.com>
Date: Wed, 4 Mar 2026 14:34:48 +0000
Subject: [PATCH] add owl rk3588 support

Signed-off-by: yifengyou <842056007@qq.com>

diff --git a/arch/arm64/boot/dts/rockchip/Makefile b/arch/arm64/boot/dts/rockchip/Makefile
index 38aaab5b2..93aef2848 100644
--- a/arch/arm64/boot/dts/rockchip/Makefile
+++ b/arch/arm64/boot/dts/rockchip/Makefile
@@ -401,6 +401,7 @@ dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3588-orangepi-5-max.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3588-orangepi-5-plus.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3588-orangepi-5-ultra.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3588-rock-5b.dtb
+dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3588-owl-ai-box-plus-v10.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3588-rock-5b-plus.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3588-rock-5-itx.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3588-rock-5t.dtb
diff --git a/arch/arm64/boot/dts/rockchip/rk3588-owl-ai-box-plus-v10.dts b/arch/arm64/boot/dts/rockchip/rk3588-owl-ai-box-plus-v10.dts
new file mode 100644
index 000000000..99ab41350
--- /dev/null
+++ b/arch/arm64/boot/dts/rockchip/rk3588-owl-ai-box-plus-v10.dts
@@ -0,0 +1,45 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+* Copyright (c) 2021 Rockchip Electronics Co., Ltd.
+*
+*/
+
+/dts-v1/;
+
+#include "rk3588-evb1-lp4.dtsi"
+#include "rk3588-evb1-imx415.dtsi"
+#include "rk3588-linux.dtsi"
+
+/ {
+	model = "OWL RK3588 AIBOX Plus V10 Board";
+	compatible = "rockchip,rk3588-evb1-lp4-v10", "rockchip,rk3588";
+
+	vbus5v_pwr: vbus5v-pwr {
+		compatible = "regulator-fixed";
+		regulator-name = "vbus5v_pwr";
+		gpio = <&gpio1 RK_PC6 GPIO_ACTIVE_HIGH>;
+		enable-active-high;
+		regulator-always-on;
+		regulator-boot-on;
+		regulator-min-microvolt = <5000000>;
+		regulator-max-microvolt = <5000000>;
+	};
+};
+
+&vcc_1v8_s0 {
+	/delete-property/ regulator-state-mem;
+	regulator-state-mem {
+		regulator-on-in-suspend;
+		regulator-suspend-microvolt = <1800000>;
+	};
+};
+
+&vcc_3v3_s0 {
+	/delete-property/ regulator-state-mem;
+	regulator-state-mem {
+		regulator-on-in-suspend;
+		regulator-suspend-microvolt = <3300000>;
+	};
+};
+
+
-- 
2.43.0

